app/assets/javascripts/admin/orders/controllers/orders_controller.js.coffee

```diff
@@ -52,3 +52,11 @@ angular.module("admin.orders").controller "ordersCtrl", ($scope, RequestMonitor,
   $scope.changePage = (newPage) ->
     $scope.page = newPage
     $scope.fetchResults(newPage)
+
+  $scope.bulkInvoice = ->
+    params = ''
+    angular.forEach $scope.selected, (selected, order_id) ->
+      params += 'order_ids[]='+order_id+'&' if selected
```

Comment:
I was trying to find a nicer way to pass this data in a request at the same time as opening a new tab, but this was the only way that worked. I tried a few other options and some workarounds using `$http.post` and manipulating new windows/tabs, but they all either failed or triggered Firefox/Chrome to block them as suspicious popups :/

Reply:
I managed to make one with 15 orders, but yes I think I'll need to pass it to a background job, then possibly have the PDF saved to the disk, and maybe have an Angular function that periodically checks to see if the file exists yet, with a spinner / loading message...